### PR TITLE
Set up the Home screen

### PR DESCRIPTION
**Pull request by yostyle**
#5

### CODE REVIEW

File: vector/src/main/java/fr/gouv/tchap/features/home/TchapHomeActivity.kt [hunk 1]
```diff
@@ -0,0 +1,427 @@
+/*
+ * Copyright 2019 New Vector Ltd
```
Comment (giomfo) [inline vector/src/main/java/fr/gouv/tchap/features/home/TchapHomeActivity.kt:2]:
```suggestion
 * Copyright 2021 New Vector Ltd
```

File: vector/src/main/java/fr/gouv/tchap/features/home/TchapHomeActivityViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,256 @@
+/*
+ * Copyright (c) 2020 New Vector Ltd
```
Comment (giomfo) [inline vector/src/main/java/fr/gouv/tchap/features/home/TchapHomeActivityViewModel.kt:2]:
```suggestion
 * Copyright (c) 2021 New Vector Ltd
```

File: vector/src/main/java/fr/gouv/tchap/features/home/TchapHomeDetailFragment.kt [hunk 3]
```diff
@@ -0,0 +1,376 @@
+/*
+ * Copyright 2019 New Vector Ltd
```
Comment (giomfo) [inline vector/src/main/java/fr/gouv/tchap/features/home/TchapHomeDetailFragment.kt:2]:
```suggestion
 * Copyright 2021 New Vector Ltd
```

File: vector/src/main/java/fr/gouv/tchap/features/home/TchapHomeDetailViewModel.kt [hunk 4]
```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright 2019 New Vector Ltd
```
Comment (giomfo) [inline vector/src/main/java/fr/gouv/tchap/features/home/TchapHomeDetailViewModel.kt:2]:
```suggestion
 * Copyright 2021 New Vector Ltd
```

File: vector/src/main/java/fr/gouv/tchap/features/home/room/list/TchapRoomListDisplayModeFilter.kt [hunk 5]
```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2019 New Vector Ltd
```
Comment (giomfo) [inline vector/src/main/java/fr/gouv/tchap/features/home/room/list/TchapRoomListDisplayModeFilter.kt:2]:
```suggestion
 * Copyright 2021 New Vector Ltd
```

File: vector/src/main/java/fr/gouv/tchap/features/home/room/list/TchapRoomListFragment.kt [hunk 6]
```diff
@@ -0,0 +1,404 @@
+/*
+ * Copyright 2019 New Vector Ltd
```
Comment (giomfo) [inline vector/src/main/java/fr/gouv/tchap/features/home/room/list/TchapRoomListFragment.kt:2]:
```suggestion
 * Copyright 2021 New Vector Ltd
```

File: vector/src/main/java/fr/gouv/tchap/features/home/room/list/TchapRoomListViewModel.kt [hunk 7]
```diff
@@ -0,0 +1,267 @@
+/*
+ * Copyright 2019 New Vector Ltd
```
Comment (giomfo) [inline vector/src/main/java/fr/gouv/tchap/features/home/room/list/TchapRoomListViewModel.kt:2]:
```suggestion
 * Copyright 2021 New Vector Ltd
```

File: vector/src/main/java/fr/gouv/tchap/features/home/room/list/TchapRoomListViewModelFactory.kt [hunk 8]
```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2019 New Vector Ltd
```
Comment (giomfo) [inline vector/src/main/java/fr/gouv/tchap/features/home/room/list/TchapRoomListViewModelFactory.kt:2]:
```suggestion
 * Copyright 2021 New Vector Ltd
```

File: vector/src/main/java/im/vector/app/core/di/ScreenComponent.kt [hunk 9]
```diff
@@ -153,6 +154,7 @@ interface ScreenComponent {
     fun inject(activity: ReAuthActivity)
     fun inject(activity: RoomDevToolActivity)
     fun inject(activity: TchapLoginActivity)
+    fun inject(activity: TchapHomeActivity)
```
Comment (giomfo) [inline vector/src/main/java/im/vector/app/core/di/ScreenComponent.kt:157]: Don't you think we should remove here HomeActivity and LoginActivity?

File: vector/src/main/java/im/vector/app/features/home/room/list/RoomListModule.kt [hunk 10]
```diff
@@ -18,10 +18,15 @@ package im.vector.app.features.home.room.list
...
+import fr.gouv.tchap.features.home.room.list.TchapRoomListViewModelFactory
 
 @Module
 abstract class RoomListModule {
 
     @Binds
     abstract fun providesRoomListViewModelFactory(factory: RoomListViewModelFactory): RoomListViewModel.Factory
+
+    @Binds
+    abstract fun providesTchapRoomListViewModelFactory(factory: TchapRoomListViewModelFactory): TchapRoomListViewModel.Factory
```
Comment (giomfo) [inline vector/src/main/java/im/vector/app/features/home/room/list/RoomListModule.kt:31]: Don't you think we should remove the previous bind related to RoomListViewModel?

File: vector/src/main/java/fr/gouv/tchap/features/home/room/list/TchapRoomListDisplayModeFilter.kt [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+ */
+
+package fr.gouv.tchap.features.home.room.list
+
+import im.vector.app.features.home.RoomListDisplayMode
+import io.reactivex.functions.Predicate
+import org.matrix.android.sdk.api.session.room.model.Membership
+import org.matrix.android.sdk.api.session.room.model.RoomSummary
+
+class TchapRoomListDisplayModeFilter(private val displayMode: RoomListDisplayMode) : Predicate<RoomSummary> {
```
Comment (giomfo) [inline vector/src/main/java/fr/gouv/tchap/features/home/room/list/TchapRoomListDisplayModeFilter.kt:24]: I don't think this class `TchapRoomListDisplayModeFilter` will be used later.
Can you confirm?
(Is it possible to keep using the existing RoomListDisplayModeFilter?)

File: vector/src/main/java/im/vector/app/features/home/room/list/RoomListModule.kt (hunk 10 above)
Comment (yostyle) [reply vector/src/main/java/im/vector/app/features/home/room/list/RoomListModule.kt:31]: this factory is injected in RoomListViewModel. Without this binding, roomListViewModelFactory is unknown and the build fails.

File: vector/src/main/java/fr/gouv/tchap/features/home/room/list/TchapRoomListDisplayModeFilter.kt (hunk 11 above)
Comment (yostyle) [reply vector/src/main/java/fr/gouv/tchap/features/home/room/list/TchapRoomListDisplayModeFilter.kt:24]: Yes I confirm this, if we have only one representation of rooms list in Tchap.

File: vector/src/main/java/im/vector/app/core/di/ScreenComponent.kt (hunk 9 above)
Comment (yostyle) [reply vector/src/main/java/im/vector/app/core/di/ScreenComponent.kt:157]: Yes but these activities implement inject method from ScreenComponent interface. So I can't

Comment (yostyle) [reply vector/src/main/java/im/vector/app/core/di/ScreenComponent.kt]: TchapHomeActivity has been replaced by TchapHomeActivity

File: vector/src/main/java/im/vector/app/features/home/room/list/RoomListModule.kt (hunk 10 above)
Comment (yostyle) [reply vector/src/main/java/im/vector/app/features/home/room/list/RoomListModule.kt]: TchapRoomListViewModelFactory has been replaced by RoomListViewModelFactory

File: vector/src/main/java/fr/gouv/tchap/features/home/room/list/TchapRoomListDisplayModeFilter.kt (hunk 11 above)
Comment (yostyle) [reply vector/src/main/java/fr/gouv/tchap/features/home/room/list/TchapRoomListDisplayModeFilter.kt]: TchapRoomListDisplayModeFilter has been replaced by RoomListDisplayModeFilter

File: vector/src/main/java/im/vector/app/features/home/room/list/RoomListViewModel.kt [hunk 12]
```diff
@@ -242,34 +242,21 @@ class RoomListViewModel @Inject constructor(initialState: RoomListViewState,
     }
 
     private fun buildRoomSummaries(rooms: List<RoomSummary>): RoomSummaries {
-        // Set up init size on directChats and groupRooms as they are the biggest ones
-        val invites = ArrayList<RoomSummary>()
```
Comment (giomfo) [inline vector/src/main/java/im/vector/app/features/home/room/list/RoomListViewModel.kt:246]: You have to keep the invites sections

**Review by giomfo — COMMENTED** [review]
SGTM, you just have to restore the invite section in RoomListViewModel

File: vector/src/main/java/im/vector/app/features/home/room/list/RoomListViewModel.kt [hunk 13]
```diff
@@ -242,34 +242,24 @@ class RoomListViewModel @Inject constructor(initialState: RoomListViewState,
     }
 
     private fun buildRoomSummaries(rooms: List<RoomSummary>): RoomSummaries {
-        // Set up init size on directChats and groupRooms as they are the biggest ones
```
Comment (giomfo) [inline vector/src/main/java/im/vector/app/features/home/room/list/RoomListViewModel.kt:245]: Sorry I just noticed this comment now. I think we should init size on `conversations` with rooms.size
Don't you think so?

Comment (yostyle) [reply vector/src/main/java/im/vector/app/features/home/room/list/RoomListViewModel.kt:245]: Yes it's a good suggestions for conversations list. Do you have any idea for this init value ?

**Review by giomfo — APPROVED** [review]
LGTM

**Pull request merged**